mundane-bytes/src/main/scala/com/ambiata/mundane/bytes/Buffer.scala

```diff
@@ -0,0 +1,26 @@
+package com.ambiata.mundane.bytes
+
+case class Buffer(bytes: Array[Byte], offset: Int, var length: Int) {
+
+  /** Ensure there is enough space in the array, and if not return a _new_ [[Buffer]] and array with the original values */
+  def allocate(i: Int): Buffer =
+    if (length + i <= bytes.length) this
+    else {
+      val b = new Buffer(new Array(math.max(bytes.length + i, bytes.length * 2)), offset, length)
+      // TODO Copy the entire array, or just from the offset?
+      System.arraycopy(bytes, 0, b.bytes, 0, bytes.length)
+      b
+    }
+}
+
+object Buffer {
+
+  def empty(initialSize: Int): Buffer =
+    new Buffer(new Array[Byte](initialSize), 0, 0)
+
+  def wrapArray(bytes: Array[Byte], offset: Int): Buffer = {
```

Comment:
Would consider adding length as an arg, or another version that allows you to do it.